### PR TITLE
Couch to SQL forms & cases docs

### PR DESCRIPTION
**Pull request by millerdev**
Add high-level readme docs to the `couch_sql_migration` app. Also improve exception message for illegal case error (very minor tweak).

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression
- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]
I'm fine with this as long as you don't see anything concerning from what I brought up about the change to the exception message

File: corehq/form_processor/backends/sql/casedb.py [hunk 1]
```diff
@@ -16,7 +16,10 @@ def __init__(self, *args, **kw):
 
     def _validate_case(self, case):
         if self.domain and case.domain != self.domain:
-            raise IllegalCaseId("Bad case id")
+            raise IllegalCaseId(
+                f"Case [{case.case_id}] in domain '{case.domain}' "
+                f"cannot be referenced by domain '{self.domain}'"
+            )
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/form_processor/backends/sql/casedb.py:22]: I think this can show up in the 'Raw Forms, Errors & Duplicates' report

**Review by snopoke — COMMENTED** [review]
Docs look good but I think the exception message need to be changed as @dannyroberts mentioned.

File: corehq/form_processor/backends/sql/casedb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/form_processor/backends/sql/casedb.py]: I removed the commit that changed the exception message. Will find a better way to reveal that information in forms & cases migration errors.

**Pull request merged**